Move checkAndUpdateWireless into VerifyPending and also handle override

naiming-zededa opened the pull request:
Signed-off-by: Naiming Shen <naiming@zededa.com>
- moved the checkAndUpdateWireless() into VerifyPending()
- leave the 'override' type of update also at the doUpdatePortConfigListAndPublish() to catch the case when the wifi override might not get into later verify stage
- change the hack of ignore zedcloud wifi null configure by copying the override wifi config into other types of update, thus now the lastresort config will have the wificonfig set by override
- tested on my device with baseimage upgrade, reboot, reboot with only wifi available

rvs commented:
This looks reasonable @naiming-zededa -- but I guess we still need to address the issue of timing between this portion of nim and other device-steps sections (this is only relevant when time is completely out of whack on boot).

I suppose we can do it in a separate PR, but it would be great to address it (unless I somehow missed how this PR addresses it).

eriknordmark commented on pkg/pillar/devicenetwork/dnc.go:
```diff
@@ -651,6 +655,11 @@ func (ctx *DeviceNetworkContext) doUpdatePortConfigListAndPublish(
 	current := getCurrentDPC(ctx) // Used to determine if index needs to change
 	currentIndex := ctx.DevicePortConfigList.CurrentIndex
 	oldConfig, _ := lookupPortConfig(ctx, *portConfig)
+	if strings.Contains(portConfig.Key, "override") {
```
Can you please put a // XXX remove this code when the controller supports WiFi config
comment so we can find it and zap it later?

eriknordmark commented on pkg/pillar/devicenetwork/dnc.go:
```diff
@@ -734,18 +736,7 @@ func checkAndUpdateWireless(ctx *DeviceNetworkContext, oCfg *types.DevicePortCon
-				break
-			}
-			// XXX
+		if oldPortCfg == nil || isOverride || !reflect.DeepEqual(oldPortCfg.WirelessCfg, pCfg.WirelessCfg) {
```
Once controller supports WiFi config, shouldn't the 'isOverride' check go away here as well? Or the whole comparison of WirelessCfg should go away? 
Please add comment indicating what should be removed once we have controller support.

eriknordmark commented on pkg/pillar/devicenetwork/dnc.go:
```diff
@@ -755,6 +746,25 @@ func checkAndUpdateWireless(ctx *DeviceNetworkContext, oCfg *types.DevicePortCon
 			}
 		}
+
+		// hack, save the override wifi param into ctx for later copying
```
// XXX remove when controller support WiFi
would make it easier to find, especially if you use the same comment string in all places where removal is needed.

naiming-zededa commented:
@eriknordmark addressed the comment on proper comments when controller supports this. Will do another testing/debug/verify later when controller is ready. There might need to be some change by then.

eriknordmark commented:
I only see the XXX addition in one place. Did you intend to add it where we need to delete code as well?

naiming-zededa commented:
Hmm, not sure what happened. my local diff
git diff fdb8ae6f9b3ac48359917c65318ab8efcad5b224 acf0bb7f50d4c10b4525d9874e3fb6beb74e25f1


showing XXX in 3 places…

> On Jan 14, 2020, at 4:54 PM, Erik Nordmark <notifications@github.com> wrote:
> 
> I only see the XXX addition in one place. Did you intend to add it where we need to delete code as well?
> 
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub <https://github.com/lf-edge/eve/pull/483?email_source=notifications&email_token=AMLHELGO3NFKEEP4IY5SAS3Q5ZNDHA5CNFSM4J7DCQOKYY3PNVWWK3TUL52HS4DFVREXG43VMVBW63LNMVXHJKTDN5WW2ZLOORPWSZGOEI6VNJA#issuecomment-574445220>, or unsubscribe <https://github.com/notifications/unsubscribe-auth/AMLHELAK62VXG4PSI23YVDTQ5ZNDHANCNFSM4J7DCQOA>.
>

naiming-zededa commented:
move this PR to #521

naiming-zededa commented:
move this PR to #521
